packages/extension/src/newtab/ShortcutLinks.tsx

```diff
@@ -29,6 +42,21 @@ export default function ShortcutLinks({
     onSaveChanges,
   } = useShortcutLinks();
 
+  const mountedRef = useRef(false);
+
+  useEffect(() => {
+    if (mountedRef.current) {
+      return;
+    }
+
+    mountedRef.current = true;
+
+    trackEvent({
+      event_name: AnalyticsEvent.Impression,
+      target_type: TargetType.Shortcuts,
+    });
+  }, [trackEvent]);
```

Comment:
So far we always just avoid the `trackEvent` I totally agree with this though.
Wondering if we should make this a custom `useOnceOffImpression` hook for now?

Just so it becomes easier to maintain if we want to expend?
Totally optional though.

Reply:
It's a tricky subject. I personally am not a fan of those because they hide implementation detail. Probelm is does not strictly mean it will fire once. It will fire each mount which is once per component lifecycle, but if component unmounts (lifecycle ends) and mounts again it will fire again. 

Also with react concepts as concurrent rendering the above might also not be "once off" as we think.

Reply:
Basically [useEffectEvent](https://react.dev/learn/separating-events-from-effects#declaring-an-effect-event) which would solve reference for `trackEvent` and the we could just use `useEffect` without dependencies.